dragontts/app/api/v1/cache.py

```
@@ -0,0 +1,292 @@
+):
+    """Cache metrics. Request metrics are date-filterable (?from=&to= YYYY-MM-DD);
+    the cache snapshot is the current point-in-time state. All from stored
+    rollups/totals — no full-table scan per call."""
+    for label, value in (("from", from_date), ("to", to_date)):
+        if value is not None:
+            try:
+                datetime.strptime(value, "%Y-%m-%d")
+            except ValueError:
+                raise HTTPException(
+                    status_code=400, detail=f"invalid {label}; use YYYY-MM-DD"
+                )
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)